src/renderless/CoreTable.vue

```diff
@@ -457,8 +457,10 @@ export default {
             if (!this.isChild(row)) {
                 const index = this.state.selected
                     .findIndex(id => id === row[this.template.dtRowId]);
-                if (!this.state.pageSelected) {
-                    this.state.selected.splice(index, 1);
+                if(!this.state.pageSelected) {
+                    if(index !== -1) {
+                        this.state.selected.splice(index, 1);
```

Comment:
this fixes the case where the selected array would be unnecessarily spliced for each unselected element

Reply:
Let's refactor the whole method to something more readable like:

```js
toggleRowSelect(row) {
    if (this.isChild(row)) {
        return;
    }

    const index = this.state.selected
        .findIndex(id => id === row[this.template.dtRowId]);

    if (index === -1 && this.state.pageSelected) {
        this.state.selected.push(row[this.template.dtRowId]);
        return;
    }

    if (index !== -1 && !this.state.pageSelected) {
        this.state.selected.splice(index, 1);
    }
},
```